*** a/Makefile	2014-06-02 12:01:51.000000000 -0500
--- b/Makefile	2014-06-17 14:47:32.246893760 -0500
***************
*** 124,130 ****
  	mkdir -p '$(INSTALL_DIR)/examples'
  	mkdir -p '$(PY_BASE_INSTALL_DIR)'
  	mkdir -p '$(INSTALL_DIR)/ups'
- 	cp ups/astrometry_net.table-dist '$(INSTALL_DIR)/ups/astrometry_net.table'
  	cp ups/astrometry_net.cfg '$(INSTALL_DIR)/ups'
  	cp __init__.py '$(PY_BASE_INSTALL_DIR)'
  	cp CREDITS LICENSE README.md '$(INSTALL_DIR)/doc'
--- 124,129 ----
