diff --git a/Makefile b/Makefile
index cd1bdfa..a0587f2 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,6 @@ install: report.txt
 	mkdir -p $(INSTALL_DIR)/python/astrometry
 	mkdir -p $(INSTALL_DIR)/python/pyfits
 	mkdir -p $(INSTALL_DIR)/ups
-	cp ups/astrometry_net.table-dist $(INSTALL_DIR)/ups/astrometry_net.table
 	cp __init__.py $(INSTALL_DIR)/python/astrometry
 	cp pyfits/*.py $(INSTALL_DIR)/python/pyfits
 	cp CREDITS GETTING-INDEXES LICENSE README $(INSTALL_DIR)/doc
